src/python/dxpy/scripts/dx.py

```diff
@@ -2513,22 +2513,18 @@ def install(args):
 
 def uninstall(args):
     app_desc = get_app_from_path(args.app)
-    if app_desc is None:
-        try:
-            dxpy.api.app_uninstall(args.app)
-            print('Uninstalled application with id: ' + args.app)
-        except:
-            try:
-                dxpy.api.app_uninstall('app-' + args.app)
-                print('Uninstalled application with name: ' + args.app)
-            except:
-                err_exit()
+    if app_desc:
+        try_call(dxpy.api.app_uninstall, app_desc['id'])
+        print('Uninstalled the {name} app'.format(name=app_desc['name']))
     else:
-        try:
-            dxpy.api.app_uninstall(app_desc['id'])
-            print('Uninstalled the ' + app_desc['name'] + ' app')
-        except:
-            err_exit()
+        user_data = dxpy.api.user_describe(dxpy.whoami(), {"fields": {"appsInstalled": True}})
+        if args.app in user_data['appsInstalled']:
+            try_call(dxpy.api.app_uninstall, 'app-' + args.app)
+        elif args.app.startwith('app-') and args.app[len('app-'):] in user_data['appsInstalled']:
+            parser.exit(1, 'Could not find the app\n')
```

Comment:
I do not understand. In L2523, we check whether the specified app is by id (e.g., app-foo) and, if so, whether the suffix (e.g., foo) is in the user describe response. I am confused why something that passes the checkpoint should result in the client throwing an exception. I think we should attempt to uninstall the app here, no?

Also, please note that L2526 is executed for a variety of reasons:
  - The specified app is by id and the id is not in the user describe response.
  - The specified app is by name and the name is not in the user describe response.
For both of the above, I think it will be better to throw an exception.